packages/block-editor/src/components/block-list/block.js

```diff
@@ -87,7 +59,6 @@ function BlockListBlock( {
 	isSelectionEnabled,
 	className,
 	name,
-	index,
```

Comment:
you can also remove the selector from withSelect

Reply:
It's part of #19597. This branch is just to illustrate the performance issue. :)